From 9bcbd7ea794c3d08c05640e426404845290cf048 Mon Sep 17 00:00:00 2001
From: Steve Best <sbest@redhat.com>
Date: Sat, 11 Oct 2014 09:28:51 -0400
Subject: [redhat] configs: disable Power7+ in-Nest cryptographic accleration for power LE

Message-id: <20141011092851.14340.1739.sendpatchset@ibm-p7r2-01.lab.bos.redhat.com>
Patchwork-id: 97478
O-Subject: [PATCH RHEL7.1 BZ1151461] redhat Kconfig: disable Power7+ in-Nest cryptographic accleration for power LE
Bugzilla: 1151461
RH-Acked-by: Don Zickus <dzickus@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

RHBZ#:
------
https://bugzilla.redhat.com/show_bug.cgi?id=1151461

Description:
------------
Disable Power7+ in-Nest cryptographic accleration driver for power LE.

When booting power LE on PowerVM the nx-crypto driver writes out numerous msgs.
Marcelo Cerri from IBM confirmed that this driver should not be built for power LE.

RHEL Version Found:
-------------------
RHEL 7.0

Brew:
-----
BE: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=8090833
LE: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=8090841

Upstream:
---------
This is a RHEL only change.

Test Status:
------------
Tested by me.

before the patch
CONFIG_CRYPTO_DEV_NX is on for both BE and LE power

[SteveBest@stevebest configs]$ grep CONFIG_CRYPTO_DEV_NX *
kernel-3.10.0-ppc64.config:CONFIG_CRYPTO_DEV_NX=y
kernel-3.10.0-ppc64.config:CONFIG_CRYPTO_DEV_NX_ENCRYPT=m
kernel-3.10.0-ppc64.config:CONFIG_CRYPTO_DEV_NX_COMPRESS=m
kernel-3.10.0-ppc64-debug.config:CONFIG_CRYPTO_DEV_NX=y
kernel-3.10.0-ppc64-debug.config:CONFIG_CRYPTO_DEV_NX_ENCRYPT=m
kernel-3.10.0-ppc64-debug.config:CONFIG_CRYPTO_DEV_NX_COMPRESS=m
kernel-3.10.0-ppc64le.config:CONFIG_CRYPTO_DEV_NX=y
kernel-3.10.0-ppc64le.config:CONFIG_CRYPTO_DEV_NX_ENCRYPT=m
kernel-3.10.0-ppc64le.config:CONFIG_CRYPTO_DEV_NX_COMPRESS=m
kernel-3.10.0-ppc64le-debug.config:CONFIG_CRYPTO_DEV_NX=y
kernel-3.10.0-ppc64le-debug.config:CONFIG_CRYPTO_DEV_NX_ENCRYPT=m
kernel-3.10.0-ppc64le-debug.config:CONFIG_CRYPTO_DEV_NX_COMPRESS=m

after the patch CONFIG_CRYPTO_DEV_NX is disabled for ppc64le.

/home/SteveBest/rhel71151461/redhat/configs
[SteveBest@stevebest configs]$ grep CONFIG_CRYPTO_DEV_NX *
kernel-3.10.0-ppc64.config:CONFIG_CRYPTO_DEV_NX=y
kernel-3.10.0-ppc64.config:CONFIG_CRYPTO_DEV_NX_ENCRYPT=m
kernel-3.10.0-ppc64.config:CONFIG_CRYPTO_DEV_NX_COMPRESS=m
kernel-3.10.0-ppc64-debug.config:CONFIG_CRYPTO_DEV_NX=y
kernel-3.10.0-ppc64-debug.config:CONFIG_CRYPTO_DEV_NX_ENCRYPT=m
kernel-3.10.0-ppc64-debug.config:CONFIG_CRYPTO_DEV_NX_COMPRESS=m
kernel-3.10.0-ppc64le.config:# CONFIG_CRYPTO_DEV_NX is not set
kernel-3.10.0-ppc64le-debug.config:# CONFIG_CRYPTO_DEV_NX is not set

---------------------------------------------------------------
Steve Best

Proposed Patch:
---------------

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/configs/generic/powerpc64le/CONFIG_CRYPTO_DEV_NX b/redhat/configs/generic/powerpc64le/CONFIG_CRYPTO_DEV_NX
new file mode 100644
index 0000000..47b9dbe
--- /dev/null
+++ b/redhat/configs/generic/powerpc64le/CONFIG_CRYPTO_DEV_NX
@@ -0,0 +1 @@
+# CONFIG_CRYPTO_DEV_NX is not set
-- 
1.7.1